### PR TITLE
DOCSUP-10113: Update the function range docs

### PR DESCRIPTION
**Pull request by atereh**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation for #7518

### CODE REVIEW

File: docs/en/sql-reference/functions/array-functions.md [hunk 1]
```diff
@@ -39,12 +39,46 @@ Accepts zero arguments and returns an empty array of the appropriate type.
...
+-   `end` - required. `UInt` number. Must be greater than `start`.
+-   `step` - optional. `UInt` number. If not specified, defaults to 1.
+
+**Returned value**
+
+-   an array of `UInt` numbers
+
+**Examples**
+
+Query with only `end` argument:
```
Comment (olgarev) [inline docs/en/sql-reference/functions/array-functions.md:67]:
```suggestion
Query with `end` argument only:
```

File: docs/en/sql-reference/functions/array-functions.md [hunk 2]
```diff
@@ -39,13 +39,43 @@ Accepts zero arguments and returns an empty array of the appropriate type.
...
+## range(end), range(\[start, \] end \[, step\]) {#range}
+
+Returns an array of `UInt` numbers from `start` to `end - 1` by `step`.
+
+**Syntax**
+``` sql
+range([start, ] end [, step])
+```
+
+An exception is thrown if query results in arrays with a total length of more than 100,000,000 elements.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/array-functions.md:52]:
```suggestion

```

File: docs/en/sql-reference/functions/array-functions.md [hunk 3]
```diff
@@ -39,13 +39,43 @@ Accepts zero arguments and returns an empty array of the appropriate type.
...
+-   `start` — The first element of the array. Optional, required if `step` is used. Default value: 0. [UInt](../data-types/int-uint.md)
+-   `end` — The number before which the array is constructed. Required. Must be greater than `start`. [UInt](../data-types/int-uint.md)
+-   `step` — Determines the incremental step between each element in the array. Optional. Default value: 1. [UInt](../data-types/int-uint.md)
+
+
+**Returned value**
+
+-   Array of `UInt` numbers from `start` to `end - 1` by `step`.
+
+
```
Comment (olgarev) [inline docs/en/sql-reference/functions/array-functions.md:66]: An exception is thrown if query results in arrays with a total length of more than 100,000,000 elements.

**Comment by atereh** [comment]
Нужен тэг can be tested, поломался Marker check

**Comment by olgarev** [comment]
> Нужен тэг can be tested, поломался Marker check

@akuzm Hi! Will you review this PR, please? And can you add the 'can be tested' label?

File: docs/en/sql-reference/functions/array-functions.md [hunk 4]
```diff
@@ -39,13 +39,44 @@ Accepts zero arguments and returns an empty array of the appropriate type.
...
+**Syntax**
+``` sql
+range([start, ] end [, step])
+```
+
+**Arguments**
+
+-   `start` — The first element of the array. Optional, required if `step` is used. Default value: 0. [UInt](../data-types/int-uint.md)
+-   `end` — The number before which the array is constructed. Required. Must be greater than `start`. [UInt](../data-types/int-uint.md)
```
Comment (tavplubix) [inline docs/en/sql-reference/functions/array-functions.md:55]: `start` can be greater or equal than `end`, but result will be empty in this case

File: docs/ru/sql-reference/functions/array-functions.md [hunk 5]
```diff
@@ -39,10 +39,49 @@ toc_title: "Массивы"
...
+
+
+**Возвращаемые значения**
+
+-   массив `UInt` чисел от `start` до `end - 1` с шагом `step`
+
+
+**Особенности реализации**
+
+-   Не поддерживаются отрицательные значения аргументов: `start`, `end`, `step` имеют формат `UInt`.
```
Comment (tavplubix) [inline docs/ru/sql-reference/functions/array-functions.md:67]:
```suggestion
-   Не поддерживаются отрицательные значения аргументов: `start`, `end`, `step` имеют тип `UInt`.
```

**Review by tavplubix — APPROVED** [review]

File: docs/en/sql-reference/functions/array-functions.md (hunk 4 above)
Comment (atereh) [reply docs/en/sql-reference/functions/array-functions.md]: fair! removed this statement in the commit below

**Pull request merged**